service/src/test/java/bio/terra/landingzone/stairway/flight/create/resource/step/CreateSentinelRunPlaybookAutomationRuleTest.java

```diff
@@ -0,0 +1,148 @@
+  @Mock AutomationRules mockAutomationRules;
+  @Mock AutomationRule.DefinitionStages.Blank mockAutomationRuleDefinitionStageBlank;
+
+  @Mock
+  AutomationRule.DefinitionStages.WithDisplayName mockAutomationRuleDefinitionStageWithDisplayName;
+
+  @Mock AutomationRule.DefinitionStages.WithOrder mockAutomationRuleDefinitionStageWithOrder;
+
+  @Mock
+  AutomationRule.DefinitionStages.WithTriggeringLogic
+      mockAutomationRuleDefinitionStageWithTriggerLogic;
+
+  @Mock AutomationRule.DefinitionStages.WithActions mockAutomationRuleDefinitionStageWithActions;
+  @Mock AutomationRule.DefinitionStages.WithCreate mockAutomationRuleDefinitionStageWithCreate;
+  @Mock AutomationRule mockAutomationRule;
+
+  private CreateSentinelRunPlaybookAutomationRule createSentinelRunPlaybookAutomationRule;
+
+  @BeforeEach
+  void setUp() {
+    createSentinelRunPlaybookAutomationRule =
+        new CreateSentinelRunPlaybookAutomationRule(
+            mockArmManagers,
+            mockParametersResolver,
+            mockResourceNameGenerator,
+            mockLandingZoneProtectedDataConfiguration);
+  }
+
+  @Test
+  void doStepSuccess() throws InterruptedException {
+    setupFlightContext(
+        mockFlightContext,
+        Map.of(
+            LandingZoneFlightMapKeys.BILLING_PROFILE,
+            new ProfileModel().id(UUID.randomUUID()),
+            LandingZoneFlightMapKeys.LANDING_ZONE_ID,
+            LANDING_ZONE_ID),
+        Map.of(
+            GetManagedResourceGroupInfo.TARGET_MRG_KEY,
+            new TargetManagedResourceGroup("mgrName", "mrgRegion"),
+            CreateLogAnalyticsWorkspaceStep.LOG_ANALYTICS_RESOURCE_KEY,
+            buildLandingZoneResource()));
+    setupArmManagersForDoStep("ruleId");
+
+    when(mockLandingZoneProtectedDataConfiguration.getLogicAppResourceId())
+        .thenReturn("logicAppResourceId");
+    when(mockLandingZoneProtectedDataConfiguration.getTenantId())
+        .thenReturn(UUID.randomUUID().toString());
+
+    StepResult stepResult = createSentinelRunPlaybookAutomationRule.doStep(mockFlightContext);
+    assertThat(stepResult, equalTo(StepResult.getStepResultSuccess()));
+    verify(mockAutomationRuleDefinitionStageWithCreate, times(1)).create();
+  }
+
+  @Test
+  void doStepMissingInputParameterThrowsException() {
+    setupFlightContext(
+        mockFlightContext,
+        Map.of(
+            LandingZoneFlightMapKeys.BILLING_PROFILE,
+            new ProfileModel().id(UUID.randomUUID()),
+            LandingZoneFlightMapKeys.LANDING_ZONE_ID,
+            LANDING_ZONE_ID),
+        Map.of());
+
+    // missing LOG_ANALYTICS_RESOURCE_KEY in workingMap
+    assertThrows(
+        MissingRequiredFieldsException.class,
+        () -> createSentinelRunPlaybookAutomationRule.doStep(mockFlightContext));
+  }
+
+  @Test
+  void undoStepSuccess() throws InterruptedException {
+    verifyUndoStep();
+  }
+
+  @Test
+  void undoStepSuccessWhenDoStepFailed() throws InterruptedException {
+    verifyUndoStep();
```

Comment:
This is identical to line 110.

Reply:
The behavior is identical in both cases now - call "deleteResource" which do nothing. I think it make sense to keep both tests in case behavior has changed. It is less likely we forget to update specific test.

Reply:
I personally find it confusing because neither test actually calls doStep first. It's really just a test of calling `undoStep` directly, and what happened in `doStep` isn't relevant.